src/bionics.cpp

```diff
@@ -67,10 +69,141 @@ const efftype_id effect_visuals( "visuals" );
 const efftype_id effect_weed_high( "weed_high" );
 
 namespace {
+typedef std::pair<body_part, size_t> cbm_pair;
+
 std::map<std::string, bionic_data> bionics;
 std::vector<std::string> faulty_bionics;
+
+// similar to content of body_part_name() but with extra entry 'Other'
+std::string bodyparts[] = { bp_asText[0], bp_asText[1], bp_asText[2], bp_asText[3],
+                            bp_asText[4], bp_asText[5], bp_asText[6], bp_asText[7],
+                            bp_asText[8], bp_asText[9], bp_asText[10], bp_asText[11],
+                            _( "Other" ), _( "All" )
+                          };
+
+void draw_background( WINDOW *win, const bool empty_list );
+void draw_header( WINDOW *win, const size_t tab_index, const std::string power_string,
+                  const std::string help_key );
+const char *power_description( std::string const &id );
+bool compare_cbm_names( cbm_pair val1, cbm_pair val2 );
+std::vector<cbm_pair> define_content( body_part bp );
+
+void draw_background( WINDOW *win, const bool empty_list )
+{
+    draw_border( win );
+    // lines below header & above footer:
+    mvwhline( win, 2, 1, 0, getmaxx( win ) - 2 );
+    mvwhline( win, getmaxy( win ) - 6, 1, 0, getmaxx( win ) - 2 );
+
+    // intersections:
+    wattron( win, BORDER_COLOR );
+    mvwaddch( win, 2, 0, LINE_XXXO );
+    mvwaddch( win, 2, getmaxx( win ) - 1, LINE_XOXX );
+    mvwaddch( win, getmaxy( win ) - 6, 0, LINE_XXXO );
+    mvwaddch( win, getmaxy( win ) - 6, getmaxx( win ) - 1, LINE_XOXX );
+    wattroff( win, BORDER_COLOR );
+
+    if( empty_list ) {
+        mvwprintz( win, 3, 2, c_ltgray, _( "No bionics installed." ) );
+    } else {
+        // titles
+        static const std::array<std::string, 3> titles = {{
+            _( "Bionic [Size]" ),
+            _( "Condition" ),
+            _( "Usage Cost" )
+        }};
+        std::array<int, 3> pos;
+            pos[0] = 2;
+            pos[1] = std::max( getmaxx( win ) / 2 - utf8_width( titles[1].c_str() ) / 2 ,
+                               pos[0] + utf8_width( titles[0].c_str() ) + 2 );
+            pos[2] = std::max( getmaxx( win ) * 2 / 3 + 7,
+                               pos[1] + utf8_width( titles[1].c_str() ) + 2);
```

Comment:
Similar: `utf8_width` also accepts a `std::string`. Remove the `.c_str()` here and above.